doc/allocators.org

```diff
@@ -12,6 +13,32 @@ This part is documented thanks to various resources:
 - [[https://www.openmymind.net/learning_zig/heap_memory/][openmymind - learning zig - heap memory]]
 - The Zig's standard library code
 
+** General pattern
+In Zig for library development, a good practice is to pass an allocator as a parameter, so the programmer can choose the best allocator for his use case.
+
+In the Zig standard library, this pattern is used a lot.
+
+For example, the ~ArrayList~ of the standard library uses this pattern :
+#+begin_src zig :imports '(std)
+    const stdout = std.io.getStdOut().writer();
+
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+
+    const intList = std.ArrayList(i32);
```

Comment:
**intList** is used just to give **list** a type or did I miss something ? 
From my understanding if I wanted to do a second array of int, I would just:
`var secondList = intList.init(gpa.allocator());`
My point is it seems weird to me to declare **intList** just so that my "real" arrays have the wanted type.
Obviously not working in zig but imo this would be better looking something like this:
`var list<i32>
list.init(gpa.allocator());`
I probably missed something there :p

Reply:
Something like this seems better to me: `var betterList = std.ArrayList(i32).init(gpa.allocator());`